client/new-stack-editor/lib/views/flexsplit/storage.coffee

```diff
@@ -0,0 +1,65 @@
+kd = require 'kd'
+FlexSplit = require './index'
+
+
+module.exports = class FlexSplitStorage extends kd.Object
+
+
+  constructor: (options = {}, data) ->
+
+    super options, data
+
+    @storage    = {}
+    @viewCount  = 0
+    @eventCount = 0
+
+    if adapterClass = @getOption 'adapter'
+      @adapter = new adapterClass
+
+
+  addView: (view, identifier, options = {}) ->
+
+    options.restore          ?= yes
+    options.keepExpandStatus ?= no # Experimental
+
+    view.on FlexSplit.EVENT_RESIZED, (fractions) =>
+      @set identifier, fractions
+      @store()
+
+    if options.keepExpandStatus
+      view.on [
+        FlexSplit.EVENT_EXPANDED
+        FlexSplit.EVENT_COLLAPSED
+      ], (fractions) =>
+
+        @eventCount++
+        @set identifier, fractions
+
+        if @eventCount is @viewCount - 1
+          @eventCount = 0
+          @store()
```

Comment:
This is currently experimental and disabled by default, what it does is a bit complicated I know :) Let's assume you have nested flex splits; one vertical and inside of that vertical two horizontal flex splits. if one of the horizontal one requests expand this will cause a chain reaction to expand outer vertical one as well (as explained here https://github.com/koding/koding/blob/fde719cba0f3c752e65769a3672b50cbfae7b484/client/new-stack-editor/lib/views/flexsplit/resizer.coffee#L34-L45) this chain reaction will also cause all the affected flex-split's to emit `Flex.EVENT_EXPANDED` (https://github.com/koding/koding/blob/fde719cba0f3c752e65769a3672b50cbfae7b484/client/new-stack-editor/lib/views/flexsplit/resizer.coffee#L56) which will cause this store to fire `store` method `n` times (`n` as times of nested splits you have) to prevent calling this much `store` and call it once when all splits expanded/collapsed I'm counting the events here and comparing them with the number of views that I've in storage.

As I said, this is experimental, it may change or removed, it's not in use currently by StackEditor.